### PR TITLE
Add helm chart with initial functional test

### PR DESCRIPTION
**Pull request by scrungus**

### CODE REVIEW

File: coral_credits/settings.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+# Build paths inside the project like this: BASE_DIR / 'subdir'.
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/5.0/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = "django-insecure--%%s+1^nkk97qur-52c0tbi4g(wxgtwuaau14etjrhvhqo0+i#"
```
Comment (JohnGarbutt) [inline coral_credits/settings.py:23]: We need to remove the settings file from here I think.

@mkjpryor can talk about how he injects the configuration, typically?

**Review by JohnGarbutt — DISMISSED** [review]

File: coral_credits/settings.py (hunk 1 above)
Comment (scrungus) [reply coral_credits/settings.py:23]: I could probably do this with values.yaml, and inject via azimuth-config: 

inject via config map: 
```
SECRET_KEY: {{ .Values.settings.secretKey | default (randAlphaNum 64) }}
```
then set in @assumptionsandg azimuth-ops role: 
```
settings:
  secretKey: "{{ azimuth_secret_key }}"
```
would probably work, unless @mkjpryor has a different suggestion?

Comment (scrungus) [reply coral_credits/settings.py:23]: I could do django-flexi-settings the same as azimuth does but I'm not sure it's required yet

**Comment by mkjpryor** [comment]
The problem with Django more generally is that the documentation assumes you have one settings file, but there are actually two classes of settings:

  1. Those that define the app and should be in the Docker image, e.g. `INSTALLED_APPS`
  2. Those that are genuinely runtime configurations, that should be injected using the chart

In Azimuth, we use https://github.com/stackhpc/django-flexi-settings to allow us to break up the settings file in this way. If we are going to stick with Django, I think it is worth doing something similar here.

File: charts/templates/clusterrole.yaml
Comment (mkjpryor) [inline charts/templates/clusterrole.yaml:1]: Why is this required? The credits app doesn't manage any Kubernetes resources does it?

File: charts/templates/clusterrolebinding.yaml
Comment (mkjpryor) [inline charts/templates/clusterrolebinding.yaml:1]: Same here

File: charts/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  selector:
+    matchLabels: {{ include "coral-credits.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels: {{ include "coral-credits.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets: {{ toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "coral-credits.fullname" . }}
```
Comment (mkjpryor) [inline charts/templates/deployment.yaml:17]: You only need a service account if you intend to create Kubernetes resources _from inside the pod_.

File: charts/templates/serviceaccount.yaml
Comment (mkjpryor) [inline charts/templates/serviceaccount.yaml:1]: Same here

File: charts/templates/settings.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+kind: Secret
+metadata:
+  name: {{ include "coral-credits.fullname" . }}
+  labels: {{ include "coral-credits.labels" . | nindent 4 }}
+type: Opaque
+# Use data because of https://github.com/helm/helm/issues/10010
+# Not doing so means that AWX-related keys are not removed on transition to the CRD
+data:
+  01-django.yaml: |
+    {{- tpl (.Files.Get "files/settings/01-django.yaml") . | b64enc | nindent 4 }}
```
Comment (mkjpryor) [inline charts/templates/settings.yaml:11]: Personally, for the number of settings you have I would just inline them here.

File: etc/coral-credits/app.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+# Database
+# https://docs.djangoproject.com/en/5.0/ref/settings/#databases
+
+DATABASES = {
+    "default": {
+        "ENGINE": "django.db.backends.sqlite3",
+        "NAME": "/data/db.sqlite3",
+    }
+}
```
Comment (mkjpryor) [inline etc/coral-credits/app.py:18]: Maybe this should be in the settings in the chart values, so it is easier to change to point at a proper DB in future?

**Review by mkjpryor — CHANGES_REQUESTED** [review]
Just a few minor comments.

File: charts/templates/settings.yaml (hunk 3 above)
Comment (scrungus) [reply charts/templates/settings.yaml:11]: might just leave it as it is as we will probably add configuration in the future

**Comment by scrungus** [comment]
@mkjpryor huh, for some reason when I first tried this I was getting some permissions errors about service accounts, which I can't remember exactly by memory. But removed it now and it still works 🤷

File: etc/coral-credits/app.py (hunk 4 above)
Comment (scrungus) [reply etc/coral-credits/app.py:18]: made database configurable via values now
